### PR TITLE
feature(sct.py): add run-pytest

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/zLsCi1KE/3407-create-an-infra-using-pytest-to-run-a-basic-set-of-operator-funcitonal-tests

Part of changes that targets scylla operator functional tests
This particular peace is addressing ability to run tests pytest-based tests

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Review by bentsi — CHANGES_REQUESTED** [review]
let's create a separate hydra command function for running operator functional tests

File: sct.py [hunk 1]
```diff
@@ -677,6 +677,29 @@ def run_test(argv, backend, config, logdir):
                   failfast=False, buffer=False, catchbreak=True)
 
 
+@cli.command('run-pytest', help="Run tests using pytest")
+@click.argument('target')
+@click.option('-b', '--backend', type=click.Choice(SCTConfiguration.available_backends), help="Backend to use")
```
Comment (vponomaryov) [inline sct.py:682]: Which backends are supported as of now?

File: sct.py [hunk 2]
```diff
@@ -677,6 +677,29 @@ def run_test(argv, backend, config, logdir):
...
+def run_pytest(target, backend, config, logdir):
+    if config:
+        os.environ['SCT_CONFIG_FILES'] = str(list(config))
+    if backend:
+        os.environ['SCT_CLUSTER_BACKEND'] = backend
+
+    if logdir:
+        os.environ['_SCT_LOGDIR'] = logdir
+
+    logfile = os.path.join(Setup.logdir(), 'output.log')
```
Comment (vponomaryov) [inline sct.py:694]: Maybe include here test run specific part? So, we could distinguish different test runs logs in one single dir.

**Review by vponomaryov — CHANGES_REQUESTED** [review]

File: sct.py (hunk 1 above)
Comment (dkropachev) [reply sct.py:682]: all of them

File: sct.py (hunk 2 above)
Comment (dkropachev) [reply sct.py:694]: It handles folders in the same manner as run-test, meaning that for each test run folder in ~/sct-results/ is going to be created.

**Review by vponomaryov — APPROVED** [review]

File: sct.py [hunk 3]
```diff
@@ -677,6 +677,29 @@ def run_test(argv, backend, config, logdir):
                   failfast=False, buffer=False, catchbreak=True)
 
 
+@cli.command('run-pytest', help="Run tests using pytest")
```
Comment (bentsi) [inline sct.py:680]: which tests this CLI command will run?

Comment (dkropachev) [reply sct.py:680]: it will run any pytest you point to sct.py run-pytest scylla_operator will run pytests from scylla_operator

**Comment by dkropachev** [comment]
Added pipeline for functional tests that is uses run-pytest command

**Review by bentsi — CHANGES_REQUESTED** [review]
let's make the longevity run pipeline code generic so it can run functional tests

**Comment by dkropachev** [comment]
> let's make the longevity run pipeline code generic so it can run functional tests

done

**Review by vponomaryov — APPROVED** [review]

**Review by bentsi — APPROVED** [review]

**Pull request merged**